### PR TITLE
feat: lauchDarkly flag added for deprecating manual flag feature

### PR DESCRIPTION
**Pull request by sumrender**
### Description
copilot:summary

copilot:poem

### Walkthrough
copilot:walkthrough

## Clickup
https://app.clickup.com/t/86cv9y0k1

## Code Coverage
Please add code coverage here

## UI Preview
LaunchDarkly Flag enabled for below organisation
https://www.loom.com/share/ae89e36d0ccd482581e146fde843e330

Flag set to false
https://www.loom.com/share/a5a541f8e72d4a95bb7e634b76711806?sid=3eaa3ddf-4718-4b85-b814-1ec45e82f3a8

### CODE REVIEW

File: src/app/core/services/launch-darkly.service.ts [hunk 1]
```diff
@@ -74,6 +74,10 @@ export class LaunchDarklyService {
     return this.getVariation('android-numeric-keypad', false);
   }
 
+  checkIfManualFlaggingFeatureIsEnabled(): Observable<boolean> {
+    return this.getVariation('deprecate_manual_flagging', false).pipe(shareReplay(1));
```
Comment (Julias0) [inline src/app/core/services/launch-darkly.service.ts:78]: shareReplay not required here

File: src/app/fyle/view-expense/view-expense.page.html [hunk 2]
```diff
@@ -19,7 +19,9 @@
         <ion-icon [src]="'../../../assets/svg/chat.svg'" slot="icon-only"></ion-icon>
       </ion-button>
 
-      <ng-container *ngIf="view === ExpenseView.team && canFlagOrUnflag$|async">
+      <ng-container
```
Comment (Julias0) [inline src/app/fyle/view-expense/view-expense.page.html:22]:
```html

<ng-container *ngIf="isManualFlaggingFeatureEnabled$|async as isManualFlaggingFeatureEnabled">
<ng-container *ngIf="canFlagOrUnflag$|async as canFlagOrUnflag">
   <ng-container *ngIf="view === ExpenseView.team && isManualFlaggingFeatureEnabled && canFlagOrUnflag">
      /// content
   </ng-container>
   </ng-container>
</ng-container>

```

File: src/app/fyle/view-expense/view-expense.page.ts [hunk 3]
```diff
@@ -59,6 +60,8 @@ export class ViewExpensePage {
 
   canFlagOrUnflag$: Observable<boolean>;
 
+  isManualFlaggingFeatureEnabled$: Observable<boolean>;
```
Comment (Julias0) [inline src/app/fyle/view-expense/view-expense.page.ts:63]: Observable<{enabled: boolean}>;

File: src/app/fyle/view-expense/view-expense.page.html (hunk 2 above)
Comment (Julias0) [reply src/app/fyle/view-expense/view-expense.page.html:22]: When observables are being loaded, you can show a skeleton loader

File: src/app/shared/components/expenses-card-v2/expenses-card.component.ts [hunk 4]
```diff
@@ -280,12 +282,15 @@ export class ExpensesCardComponent implements OnInit {
     this.isPerDiem = this.category === 'per diem';
 
     this.isDraft = this.sharedExpenseService.isExpenseInDraft(this.expense);
-    this.isPolicyViolated = this.expense.is_manually_flagged || this.expense.is_policy_flagged;
     this.isCriticalPolicyViolated = this.sharedExpenseService.isCriticalPolicyViolatedExpense(this.expense);
     this.vendorDetails = this.sharedExpenseService.getVendorDetails(this.expense);
     this.expenseFieldsService.getAllMap().subscribe((expenseFields) => {
       this.expenseFields = expenseFields;
     });
+    this.launchDarklyService.checkIfManualFlaggingFeatureIsEnabled().subscribe((ldFlag) => {
```
Comment (Julias0) [inline src/app/shared/components/expenses-card-v2/expenses-card.component.ts:290]: dont resolve observables inside presentational components. Resolve in parent and pass via input

File: src/app/shared/components/reports-card/reports-card.component.ts [hunk 5]
```diff
@@ -28,14 +29,20 @@ export class ReportsCardComponent implements OnInit {
...
+  constructor(private launchDarklyService: LaunchDarklyService) {}
 
   ngOnInit() {
     this.showDate =
       (this.erpt && new Date(this.erpt.rp_created_at).toDateString()) !==
       (this.prevDate && new Date(this.prevDate).toDateString());
 
     this.reportCurrencySymbol = getCurrencySymbol(this.erpt.rp_currency, 'wide');
+
+    this.launchDarklyService.checkIfManualFlaggingFeatureIsEnabled().subscribe((ldFlag) => {
```
Comment (Julias0) [inline src/app/shared/components/reports-card/reports-card.component.ts:43]: dont resolve observables inside presentational components. Resolve in parent and pass via input

**Review by howdysuraj — APPROVED** [review]

**Pull request merged**